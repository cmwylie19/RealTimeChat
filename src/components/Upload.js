
import React, { Fragment } from 'react'
import { Button, Avatar } from '@patternfly/react-core';
import { PlusCircleIcon } from '@patternfly/react-icons'
import './upload.css'

export default function Upload(props) {
    const {toggleEdit, edit, setAvatar, avatar } = props;
    return (
        <Fragment>
            <div className="upload-btn-wrapper">
                {!edit && <Avatar src={avatar} alt="Avatar image" />}
                {edit && 
                <Fragment>
                <Button variant="link" icon={<PlusCircleIcon />}>Upload </Button>
                <input type="file" onChange={event => toggleEdit(setAvatar(URL.createObjectURL(event.target.files[0])))} />
                </Fragment>}
            </div>
        </Fragment>
    )
}
