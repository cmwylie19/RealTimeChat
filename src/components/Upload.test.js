import 'raf/polyfill';
import * as React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Upload from './Upload';

configure({ adapter: new Adapter() });

describe('Upload',()=>{
    test('Upload component has the correct css class', () => {
        const view = shallow( <Upload title="test" /> );
        expect(view.props().className).toMatch(/upload-btn-wrapper/);
      });

    test('matches snapshot',()=>{
        const view = mount(<Upload />);
        expect(view).toMatchSnapshots()
    })
})
