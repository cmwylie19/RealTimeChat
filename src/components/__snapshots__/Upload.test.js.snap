// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="upload-btn-wrapper"
>
  <img
    alt="Avatar image"
    className="pf-c-avatar"
    src=""
  />
</div>
`;
